src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -101,4 +106,58 @@ private CallbackValidations() {
     static Validation<String, List<Map<String, Object>>> hasAScannedRecord(CaseDetails theCase) {
         return scannedDocumentValidator.validate(theCase);
     }
+
+    @Nonnull
+    static Validation<String, Void> hasValidCombinationOfEventIdAndClassification(
+        CaseDetails theCase,
+        String eventId
+    ) {
+        return getJourneyClassification(theCase)
+            .map(classification -> {
+                    switch (classification) {
+                        case SUPPLEMENTARY_EVIDENCE:
+                            return ATTACH_TO_EXISTING_CASE.equalsIgnoreCase(eventId)
+                                ? Validation.<String, Void>valid(null)
+                                : Validation.<String, Void>invalid(
+                                format("The %s event is not supported for %s", eventId, classification
+                                )
+                            );
+                        case EXCEPTION:
+                            // When classification is exception and case data has ocr in it and
+                            // if orchestrator url is configured for create case it will be invalid CCD configuration.
+                            boolean isExceptionRecordWithOcr = exceptionRecordHasOcr(theCase);
+                            return !isExceptionRecordWithOcr && ATTACH_TO_EXISTING_CASE.equalsIgnoreCase(eventId)
+                                ? Validation.<String, Void>valid(null)
+                                : Validation.<String, Void>invalid(
+                                errorMessage(eventId, classification, isExceptionRecordWithOcr)
+                            );
+                        default:
+                            return Validation.<String, Void>invalid(
+                                format("Invalid journey classification %s", classification)
+                            );
+                    }
+                }
+            ).orElseGet(() -> invalid("No journey classification supplied"));
+    }
+
+    private static String errorMessage(String eventId, String classification, boolean isExceptionRecordWithOcr) {
```

Comment:
Worth including some more information in the method name, so that it's clear what kind of error messages it creates.

Reply:
errorMessageBasedOnOcrPresence ?